### ISSUE TITLE
🐛 Bug in CSS parser for Tailwind: custom variants with `()` syntax not supported

### ISSUE THREAD

**Issue by PaulRBerg** [biomejs/biome#7964]
### Environment information

```block
Biome v2.3.2
```

### What happened?

Biome is unable to parse this Tailwind custom variant:

```css
@custom-variant max-xxs (@media (max-width: 320px));
```

Getting this error:

> Expected a selector but instead found '@'.

But this is valid syntax according to the [Tailwind docs](https://tailwindcss.com/docs/adding-custom-styles#adding-custom-variants).

[**Playground link**](https://biomejs.dev/playground/?cssModules=true&code=QABjAHUAcwB0AG8AbQAtAHYAYQByAGkAYQBuAHQAIABtAGEAeAAtAHgAeABzACAAKABAAG0AZQBkAGkAYQAgACgAbQBhAHgALQB3AGkAZAB0AGgAOgAgADMAMgAwAHAAeAApACkAOwAgAC8AKgAgAC0ALQBiAHIAZQBhAGsAcABvAGkAbgB0AC0AeAB4AHMAIAAqAC8A&language=css)

### Expected result

Biome should be able to parse this Tailwind syntax.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
According to the docs, it looks like you should be using this syntax:

```css
@custom-variant any-hover {
  @media (any-hover: hover) {
    &:hover {
      @slot;
    }
  }
}
```

AFAIK, what's in the parenthesis in the shorthand version is any CSS selector, of which `@media` is not.

Can you find any other docs for this?

**Comment by PaulRBerg** [comment]
hmm, you might be right @dyc3. I need to double-check this, though, as I think my syntax is working (but maybe I'm fooling myself somehow).

**Comment by PaulRBerg** [comment]
Looks like the syntax is valid albeit not being explicitly documented in the Tailwind v4 docs.

<img width="248" height="97" alt="Image" src="https://github.com/user-attachments/assets/c4e98d96-8c25-4e4b-917b-908b2a7bf231" />

<img width="225" height="146" alt="Image" src="https://github.com/user-attachments/assets/b4c8f73f-002b-4090-8c96-7dc863450811" />

`min-lg` is used like this in HTML:

```
<div className="min-lg:text-green-500">
  min-lg (≥992px): <span className="min-lg:hidden">❌ Not active</span>
  <span className="hidden min-lg:inline">✅ ACTIVE</span>
</div>
```

**Comment by ematipico** [comment]
We should check the tests

**Comment by fmajestic** [comment]
Not sure if I should open a separate issue, there's a similar problem with a selector list (comma):

```css
@custom-variant cell (th:has(&), td:has(&));
/*                             ^ Biome: expected `)` but instead found `,` (parse) */
```

This is valid and e.g. `cell:w-[220px]` compiles to:

```css
.cell\:w-\[220px\] {
    th:has(&), td:has(&) {
        width: 220px;
    }
}
```

A workaround is to wrap the selector list in a `:where` or `:is`: `@custom-variant cell (:is(th:has(&), td:has(&)));`, or use the expanded form with curly braces and `@slot;`.

[Playground link](https://biomejs.dev/playground/?code=QABjAHUAcwB0AG8AbQAtAHYAYQByAGkAYQBuAHQAIABjAGUAbABsACAAKAA6AHcAaABlAHIAZQAoAHQAaAA6AGgAYQBzACgAJgApACwAIAB0AGQAOgBoAGEAcwAoACYAKQApACkAOwAKAAoAQABjAHUAcwB0AG8AbQAtAHYAYQByAGkAYQBuAHQAIABjAGUAbABsACAAKAA6AGkAcwAoAHQAaAA6AGgAYQBzACgAJgApACwAIAB0AGQAOgBoAGEAcwAoACYAKQApACkAOwAKAAoAQABjAHUAcwB0AG8AbQAtAHYAYQByAGkAYQBuAHQAIABjAGUAbABsACAAewAKACAAIAB0AGgAOgBoAGEAcwAoACYAKQAsACAAdABkADoAaABhAHMAKAAmACkAIAB7AAoAIAAgACAAIABAAHMAbABvAHQAOwAKACAAIAB9AAoAfQAKAAoAQABjAHUAcwB0AG8AbQAtAHYAYQByAGkAYQBuAHQAIABjAGUAbABsACAAKAB0AGgAOgBoAGEAcwAoACYAKQAsACAAdABkADoAaABhAHMAKAAmACkAKQA7AAoA&language=css)

**Comment by dyc3** [comment]
@fmajestic I think technically its a separate issue, but its so closely related to this that it's fine to just have it here. Thanks for bringing it up.

**Comment by dyc3** [comment]
I was able to find a test in the tailwind repo for the `@media` syntax: https://github.com/tailwindlabs/tailwindcss/blob/4363426384d98c3545491738b26afd6a0c03e22c/packages/tailwindcss/src/intellisense.test.ts#L420

It's loose enough to accept any possible at-rule, not just `@media`.
